pkg/gui/keybindings.go

```diff
@@ -28,178 +32,192 @@ type Binding struct {
+	"<c-_>":       {gocui.KeyCtrlUnderscore, gocui.ModCtrl},
+	"<backspace>": {gocui.KeyBackspace, gocui.ModNone},
+	"<tab>":       {gocui.KeyTab, gocui.ModNone},
+	"<backtab>":   {gocui.KeyBacktab, gocui.ModNone},
+	"<enter>":     {gocui.KeyEnter, gocui.ModNone},
+	"<a-enter>":   {gocui.KeyEnter, gocui.ModAlt},
+	"<esc>":       {gocui.KeyEsc, gocui.ModNone},
+	"<space>":     {gocui.KeySpace, gocui.ModNone},
+	"<f1>":        {gocui.KeyF1, gocui.ModNone},
+	"<f2>":        {gocui.KeyF2, gocui.ModNone},
+	"<f3>":        {gocui.KeyF3, gocui.ModNone},
+	"<f4>":        {gocui.KeyF4, gocui.ModNone},
+	"<f5>":        {gocui.KeyF5, gocui.ModNone},
+	"<f6>":        {gocui.KeyF6, gocui.ModNone},
+	"<f7>":        {gocui.KeyF7, gocui.ModNone},
+	"<f8>":        {gocui.KeyF8, gocui.ModNone},
+	"<f9>":        {gocui.KeyF9, gocui.ModNone},
+	"<f10>":       {gocui.KeyF10, gocui.ModNone},
+	"<f11>":       {gocui.KeyF11, gocui.ModNone},
+	"<f12>":       {gocui.KeyF12, gocui.ModNone},
+	"<insert>":    {gocui.KeyInsert, gocui.ModNone},
+	"<delete>":    {gocui.KeyDelete, gocui.ModNone},
+	"<home>":      {gocui.KeyHome, gocui.ModNone},
+	"<end>":       {gocui.KeyEnd, gocui.ModNone},
+	"<pgup>":      {gocui.KeyPgup, gocui.ModNone},
+	"<pgdown>":    {gocui.KeyPgdn, gocui.ModNone},
+	"<up>":        {gocui.KeyArrowUp, gocui.ModNone},
+	"<down>":      {gocui.KeyArrowDown, gocui.ModNone},
+	"<left>":      {gocui.KeyArrowLeft, gocui.ModNone},
+	"<right>":     {gocui.KeyArrowRight, gocui.ModNone},
+	"<c-right>":   {gocui.KeyArrowRight, gocui.ModCtrl},
 }
 
 func (gui *Gui) getKeyDisplay(name string) string {
 	key := gui.getKey(name)
 	return GetKeyDisplay(key)
 }
 
-func GetKeyDisplay(key interface{}) string {
+func GetKeyDisplay(keyMod KeyMod) string {
 	keyInt := 0
 
-	switch key := key.(type) {
+	value, ok := keyMapReversed[keyMod]
+	if ok {
+		return value
+	}
+
+	// TODO: modifier
+	switch key := keyMod.Key.(type) {
 	case rune:
 		keyInt = int(key)
 	case gocui.Key:
-		value, ok := keyMapReversed[key]
-		if ok {
-			return value
-		}
 		keyInt = int(key)
 	}
 
 	return fmt.Sprintf("%c", keyInt)
 }
 
-func (gui *Gui) getKey(key string) interface{} {
+func (gui *Gui) getKey(key string) KeyMod {
+	mod := gui.getModifier(key)
 	runeCount := utf8.RuneCountInString(key)
 	if runeCount > 1 {
-		binding := keymap[strings.ToLower(key)]
-		if binding == nil {
+		binding, found := keymap[strings.ToLower(key)]
+		if !found {
 			log.Fatalf("Unrecognized key %s for keybinding. For permitted values see %s", strings.ToLower(key), constants.Links.Docs.CustomKeybindings)
 		} else {
 			return binding
 		}
 	} else if runeCount == 1 {
-		return []rune(key)[0]
+		return KeyMod{[]rune(key)[0], mod}
 	}
 	log.Fatal("Key empty for keybinding: " + strings.ToLower(key))
-	return nil
+	return KeyMod{nil, gocui.ModNone}
+}
+
+func (gui *Gui) getModifier(key string) gocui.Modifier {
+	runeCount := utf8.RuneCountInString(key)
+	if runeCount > 1 {
+		if []rune(key)[1] == 'c' {
```

Comment:
could we add a comment explaining this

Reply:
Oops, this supposed to be only a quick hack for testing and isn't actually needed now. I'll delete this.

Reply:
haha all good, my bad for being so fine-grained when you said it was a WIP in the description